### ISSUE TITLE
"State" section of Properties window doesn't always show up.

### ISSUE THREAD

**Issue by xouqoa** [rangle/augury#1114]
Augury version (required): 1.7
Angular version (required): 4.1.1
Date: 5/16/2017
OS: Windows 10

This is more of a general question than an actual issue, perhaps.

Some of our components don't show the State section of the properties window. It worked in the past, and then at some point it stopped. I'm not sure if this is a code change thing, or an actual issue with Augury. 

I guess what I'm trying to find out is if there are any common code pitfalls that can cause this issue or not. In one parent component, we have a child component that doesn't show anything, and a separate child component that functions as expected. It's kinda weird and I'm just not sure what to look for/at.

**Comment by JanEggers** [comment]
same issue here,

it would be nice that in case augury is not able to display the state to generate a logmessage in the console or in the area the state would normally show up. that way we could provide better feedback whats not working or fix the issue in that component

**Comment by colthreepv** [comment]
I am having the same issue with version **1.9**
Are you sure your version is 1.7? Chrome extensions should auto-update

**Comment by xouqoa** [comment]
I uninstalled/installed to get 1.9 -- it wasn't doing it automatically (Chrome Canary). Problem seems to be resolved for me at least. I'll keep an eye on it.

**Comment by xouqoa** [comment]
Still having issues with the State section not showing up. Would this have anything to do with lazy loaded modules, maybe?

Also, with the 1.9 update, I have to close the F12 tools if I reload my page or I get the "this is not an angular application" message in Augury. Separate issue I'm sure, but figured I'd mention it.

**Comment by AlrikOlson** [comment]
I noticed when we had a property that contained a "." in the name that the State section stopped showing up. As soon as we replaced the "." in the property, the State section showed up again.

This does not work:
`
var options = {
    ...
    ...
    events: {
        'froala.editor': true
    }
}
`

This does work:
`
var options = {
    ...
    ...
    events: {
        'froalaeditor': true
    }
}
`

**Comment by mhengelein** [comment]
Still having this issue with 1.15.0. :(

**Comment by jacobedawson** [comment]
I'm having this issue with 1.15.0 / Angular 5- not having the state available greatly reduces the use cases for this extension...

**Comment by jainvishal520** [comment]
Any updates on this? I'm not able to see component states!

**Comment by sunstar007** [comment]
Same here - I don't see the state on many of the components in our Angular 5 application (works on some components though). There's no error message to indicate what could be wrong, and the components are very similar in the way they're structured

**Comment by kpodkalicki** [comment]
Same here - no state of components visible for Angular 5 app.

**Comment by lautaros86** [comment]
Same here  - no state of components visible for Angular 6 app.
Windows 10

**Comment by santiago-elustondo** [comment]
The bug described in #1283 is confirmed: Injecting a router into a component causes the disappearance of the state display.
Is anyone else also injecting `Router` into their components?  As in: 
```
constructor( private router: Router ) {}
```

If not, are you injecting anything else? (`HttpClient` has also been reported to cause this, although it works for me)

**Comment by MattWilliamsDev** [comment]
@santiago-elustondo I am injecting `ActivatedRoute` in mine but not Router and having no luck seeing state.

**Comment by santiago-elustondo** [comment]
Hey, thanks @MattWilliamsDev.

Is the state missing for all the components, or only those that inject `ActivatedRoute`?

**Comment by lautaros86** [comment]
Hi!
i'm injecting HttpClient at some services, Router at some Components and ActivatedRoute in some other. But the state is missing for all components of app, even if I have no injection.

**Comment by santiago-elustondo** [comment]
@lautaros86, are you using version 1.16.1  of the [Canary Build](https://chrome.google.com/webstore/detail/augury-canary-build/leechohfifidanmkioncamdcibhmmcji) ?

**Comment by s001dxp** [comment]
I have Angular Version: 4.4.7
Augury 1.16.2

And do not see state on any of my components. BTW, it only worked sporadically in recent versions.

On one component, for example, I'm injecting custom built services and the StateService from ui-router.

**Comment by slavede** [comment]
I have Angular 5.2.3 and Augury 1.16.3. There are no states displayed :(

**Comment by lukeatron** [comment]
Angular 4.4.6, Augury 1.16.3.  No state on 90% of the components.  The issue is definitely not resolved.

I can see the property values of the component in the console via `$$el.context` just fine.

**Comment by santiago-elustondo** [comment]
Two new edge cases causing this issue have been identified and fixes in the new version of the [Canary Build](https://chrome.google.com/webstore/detail/augury-canary-build/leechohfifidanmkioncamdcibhmmcji) (version 1.17.1)

@lukeatron, @slavede, @s001dxp, can you confirm / disconfirm that version 1.17.1 of the [Canary Build](https://chrome.google.com/webstore/detail/augury-canary-build/leechohfifidanmkioncamdcibhmmcji) eliminates the issue for you?

**Comment by slavede** [comment]
@santiago-elustondo yes, it eliminates the issue for me. Thanks!

**Comment by lukeatron** [comment]
@santiago-elustondo I am now seeing the state for (I believe but haven't checked exhaustively) all components.  It takes a little bit to load and display the state when I click on a component in the tree.  Everything but the state shows immediately but while the state is loading the browser is unresponsive and loading can take 10s of seconds for more complex components with a lot of properties.  

Also, mousing over the component tree causes the browser to go unresponsive for a couple seconds before it shows the hover highlight.  It seems like there's some lengthy work being done in a synchronous context that's bogging down the browser (Chrome 66.0.3359.181 in my case).

**Comment by santiago-elustondo** [comment]
Canary merged with master as official 1.18 release. 
Please reopen if this issue persists.

**Comment by AdJesumPerMariam** [comment]
I deactivated all of my extensions before posting. I am still experiencing the issue on both master and the Canary Build. I always see "No state to show". Angular 6.0.2, Chrome 65.0.3325.181, Fedora 27 

![image](https://user-images.githubusercontent.com/32486009/40686610-e4df84cc-6365-11e8-9542-dd1003bb808e.png)

**Comment by lautaros86** [comment]
Its fixed for me!

**Comment by lukeatron** [comment]
I finally got a chance to test this.  It's mostly working except for a few components that have a lot of public members that show up in the state pane.  For these components the state pane shows "Loading component state..." for about 10 seconds then switches to "No state to show".  It seems like it's timing out and giving up.

To be fair, I'm forced to run Norton AV on my machine by work policy and it absolutely butchers the performance.  My gut feeling is that if the browser was running like it should, this wouldn't be an issue.  

None the less, it would be nice if the state pane could be progressively loaded as things are resolved and maybe give up on particular fields if they're taking to long (perhaps with the option to lazy load those ones on demand if you really want to wait).  Not getting any state info on these big crappy components (where most of the bugs live of course) really hampers the usefulness of the tool.

**Comment by AdJesumPerMariam** [comment]
Can this issue be reopened ?

**Comment by santiago-elustondo** [comment]
Thank you for the info, @lukeatron.

We are aware that this issue persists for some users. 

We are currently attempting a fix-all solution.
An example repository would be useful to help us figure out the outstanding cases that cause this problem.

**Comment by lukeatron** [comment]
The only place I'm seeing this behavior is in a poorly architected internal company application that I obviously can't share.  After some more use I'm finding the problem is pretty pervasive and affects some components that I wouldn't expect to be problematic based on the number of properties alone.  Perhaps some of these components have properties with very deep trees that need to be walked.

It does seem like loading is being cutoff by timeout as the message switches to "No state to show" at almost exactly 10 seconds every time.  Are there any configuration variables for the timeout that I could mess with or is it hard coded to 10 seconds?

**Comment by AdJesumPerMariam** [comment]
@santiago-elustondo I was able to reproduce the issue in this repository: https://github.com/AdJesumPerMariam/augryissue1114

**Comment by santiago-elustondo** [comment]
Thank you very much, @AdJesumPerMariam.
Will look at this today.

**Comment by AdJesumPerMariam** [comment]
Hi @santiago-elustondo ,

I installed 1.18.1 on canary and can see that the issue I have had with my project with the state being empty was fixed! Thank you very much!

The only issue I have now is that the component tree does not seem to be automatically updating as I navigate through the application. I have to close the console and open it again to be able to get the component tree of the current application state. I reverted back to 1.18.0 and this behavior exists on the Angular Tour of Heroes example. Let me know if you would like me to create another issue for this, or if this is expected behavior.

Thanks again!
